File: assignments/credit/javascript/credit_test_runner.js
const assert = require('assert');
const { Credit } = require('./credit.js'); // Assuming student file is Credit.js

function runTests() {
    const solver = new Credit();
    // Test Cases: { number: string|null, expected: string }
    const testCases = [
        { number: "4062901840127322", expected: "VISA" }, { number: "4000000000000", expected: "VISA" },
        { number: "378282246310005", expected: "AMEX" }, { number: "340000000000000", expected: "AMEX" },
        { number: "5105105105105100", expected: "MASTERCARD" }, { number: "5555555555555555", expected: "MASTERCARD" },
        { number: "4062901840127321", expected: "INVALID" }, { number: "378282246310006", expected: "INVALID" },
        { number: "5105105105105101", expected: "INVALID" }, { number: "411122223333", expected: "INVALID" },
        { number: "41112222333344445", expected: "INVALID" }, { number: "6011000000000000", expected: "INVALID" },
        { number: "1234567890123", expected: "INVALID" }, { number: null, expected: "INVALID" },
        { number: "", expected: "INVALID" }, { number: "abcde12345abcde", expected: "INVALID" }
    ];

    let passedCount = 0;
    let report = [];
    console.log("===== Starting Automated Tests for Credit Card Validator (JavaScript) =====");

    testCases.forEach((tc, i) => {
        let actual = "Error";
        const inputDesc = tc.number === null ? "null" : `"${tc.number}"`;
        const caseIdentifier = `Test Case #${i + 1}: Input ${inputDesc}`;
        try {
            actual = solver.validateCreditCard(tc.number);
            assert.strictEqual(actual, tc.expected);
            report.push(`${caseIdentifier} -> ✅ PASS`);
            passedCount++;
        } catch (error) {
             report.push(`${caseIdentifier} -> ❌ FAIL/ERROR: Expected "${tc.expected}", but got "${actual}"`);
        }
    });

    console.log(report.join('\n\n'));
    console.log("\n===== Test Summary =====");
    console.log(`Total Tests: ${testCases.length}`); console.log(`Passed: ${passedCount}`); console.log(`Failed: ${testCases.length - passedCount}`); console.log("========================");
    console.log(`[SUMMARY]:${passedCount}:${testCases.length}`);
}

runTests();

File: assignments/credit/python/credit_test_runner.py
import sys
from credit import Credit # Import the student's class

def run_tests():
    solver = Credit()
    # Test Cases: (cardNumber, expectedResult)
    test_cases = [
        ("4062901840127322", "VISA"), ("4000000000000", "VISA"),
        ("378282246310005", "AMEX"), ("340000000000000", "AMEX"),
        ("5105105105105100", "MASTERCARD"), ("5555555555555555", "MASTERCARD"),
        ("4062901840127321", "INVALID"), ("378282246310006", "INVALID"),
        ("5105105105105101", "INVALID"), ("411122223333", "INVALID"),
        ("41112222333344445", "INVALID"), ("6011000000000000", "INVALID"),
        ("1234567890123", "INVALID"), (None, "INVALID"), ("", "INVALID"),
        ("abcde12345abcde", "INVALID")
    ]
    passed_count = 0
    report = []
    print("===== Starting Automated Tests for Credit Card Validator (Python) =====")

    for i, (input_num, expected) in enumerate(test_cases):
        actual = "Error"
        case_identifier = f'Test Case #{i + 1}: Input "{input_num}"'
        try:
            actual = solver.validate_credit_card(input_num)
            report.append(case_identifier)
            report.append(f' -> Expected: "{expected}", Got: "{actual}"')

            if actual == expected:
                report.append(" -> ✅ PASS\n")
                passed_count += 1
            else:
                report.append(" -> ❌ FAIL\n")
        except Exception as e:
            report.append(case_identifier)
            report.append(f" -> ❌ ERROR: An exception occurred - {type(e).__name__}\n")

    print("\n".join(report))
    print("===== Test Summary ====="); print(f"Total Tests: {len(test_cases)}"); print(f"Passed: {passed_count}"); print(f"Failed: {len(test_cases) - passed_count}"); print("========================")
    print(f"[SUMMARY]:{passed_count}:{len(test_cases)}")

if __name__ == "__main__":
    sys.path.append('.')
    run_tests()